Signal/src/util/OWSBackupExportJob.m

```diff
@@ -211,310 +369,297 @@ - (void)configureExport:(OWSBackupJobBoolCompletion)completion
-                                                 TSMessage *message = object;
-                                                 if (message.isExpiringMessage) {
-                                                     return;
-                                                 }
-                                             }
-                                             TSInteraction *interaction = object;
-                                             // Ignore dynamic interactions.
-                                             if (interaction.isDynamicInteraction) {
-                                                 return;
-                                             }
-                                             [interaction saveWithTransaction:dstTransaction];
-                                             copiedInteractions++;
-                                             copiedEntities++;
-                                         }];
-
-            // Copy migrations.
-            [srcTransaction
-                enumerateKeysAndObjectsInCollection:[OWSDatabaseMigration collection]
-                                         usingBlock:^(NSString *key, id object, BOOL *stop) {
-                                             if (self.isComplete) {
-                                                 *stop = YES;
-                                                 return;
-                                             }
-                                             if (![object isKindOfClass:[OWSDatabaseMigration class]]) {
-                                                 OWSProdLogAndFail(
-                                                     @"%@ unexpected class: %@", self.logTag, [object class]);
-                                                 return;
-                                             }
-                                             OWSDatabaseMigration *migration = object;
-                                             [migration saveWithTransaction:dstTransaction];
-                                             copiedMigrations++;
-                                             copiedEntities++;
-                                         }];
-        }];
+        copiedMigrations = exportEntities(transaction,
+            [OWSDatabaseMigration collection],
+            [OWSDatabaseMigration class],
+            nil,
+            OWSSignalServiceProtosBackupSnapshotBackupEntityTypeMigration);
     }];
 
-    if (self.isComplete) {
+    if (aborted || self.isComplete) {
         return NO;
     }
-    // TODO: Should we do a database checkpoint?
 
-    DDLogInfo(@"%@ copiedThreads: %llu", self.logTag, copiedThreads);
-    DDLogInfo(@"%@ copiedMessages: %llu", self.logTag, copiedInteractions);
-    DDLogInfo(@"%@ copiedEntities: %llu", self.logTag, copiedEntities);
-    DDLogInfo(@"%@ copiedAttachments: %llu", self.logTag, copiedAttachments);
-    DDLogInfo(@"%@ copiedMigrations: %llu", self.logTag, copiedMigrations);
+    @autoreleasepool {
```

Comment:
I was probably overly liberal with my usage of `@autoreleasepool` but I'm trying to future-proof this work and ensure that the (often expensive) memory usage of the import/export process is cleaned up ASAP.